### PR TITLE
Meralyne

### PR DESCRIPTION
**Pull request by Lainwright-Lazuli**
## About The Pull Request

Nerfs Meralyne as part of a planned medical overhaul by adding a slowdown debuff by around 100%. It increments speed (i.e. M.reagent_move_delay_modifier) by 1.0, resulting in a 2 second slowdown over a 12 second, 5 lap, ~15 tile test run.

## Why It's Good For The Game

The nerf is part of a planned future overhaul to give kelotane/bicaridine more of a longer-term, better healing role while having patches implemented for faster, but smaller healing. This keep Meralyne's potency while reducing its effectiveness as a combat chem: making it more for situational non-combat effectiive healing and giving both the planned patches and kelobica a more combat-focused role.

## Changelog
:cl:
balance: Gave Meralyne a mild slowdown effect as part of its duration, lending it more for non-combat usage.
/:cl:

### CODE REVIEW

**Comment by Hughgent** [comment]
ooof, that move delay is pretty strong. 1 is HIGH. check out the armor slowdown numbers.

**Comment by Lainwright-Lazuli** [comment]
I've tested it in-game, a person with shoes could still go relatively fast. Should I drop it to 0.5 or less?

**Comment by Hughgent** [comment]
I'm saying that it would probably stack with any other slowdown modifiers. play around with the regular armor and see what feels best for you.

**Comment by Lainwright-Lazuli** [comment]
Certainly, I shall.

**Comment by Lainwright-Lazuli** [comment]
Tested with several brands of armor. On a personally subjective viewpoint, I find that the tweaks work well without screwing up either the intended goal of opening the way for the future medical overhaul, or making healed marines move like molasses. However, I would not mind feedback based on the numbers I have calculated:

**Current Stats with Meralyne Nerf:**

Unarmored, 10 seconds for 55, about 2 per lap. No pill.
Unarmored, 15 seconds for 55, about 3 per lap. Pill.
Edge armor, 15 seconds for 55. around 2.8 per lap. No pill.
Edge armor, 20 seconds for 55. around 4 per lap. Pill.
IS armor, 15 seconds for 55. around 3 per lap. No pill.
IS armor, 20 seconds for 55. around 4 per lap. Pill.
LB armor, 12 seconds for 55, around 2.5 per lap. No pill.
LB armor, 17 seconds for 55, around 3.2 per lap. Pill.
H armor, 17 seconds for 55, around 3.2 per lap. No pill.
H armor, 20 seconds for 55, around 4 per lap. Pill.
Preliminary tests done over 55 tiles, with a switch back per 11 tiles. 5 laps, 11 tiles each.

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 1]
```diff
@@ -596,9 +596,8 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
 	scannable = TRUE
 
 /datum/reagent/medicine/meralyne/on_mob_life(mob/living/M, alien)
-	. = ..()
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]: This will make it not metabolize

**Comment by AdamElTablawy** [comment]
as an FYI noncombat healing only chems are sort of pointless because when someone is out of the fight they're out of the fight and it'll take them a fair amount of time to wade back _into_ the fight.

sort of why nobody uses hypervene, for example. if you're out of the fight you just wait off the neuro, that's basically it. same deal with this-if you're out of the fight ordinary AFKs and bicardine will be sufficient to heal you up, and regardless at that point if it's anything more than a fracture you're probably going up anyway.

so meralyne basically just won't get produced as a result because it's now bad for in-combat healing and irrelevant otherwise since out of combat healing _speed_ is irrelevant and nobody will go to the effort of making it.

**Comment by Rohesie** [comment]
Stacking combat healing is an issue, though. We'll have to see if the cost is too high, but this is intended to be more of a doctor or backline medic tool than a frontline one.

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 2]
```diff
@@ -627,6 +627,27 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
...
+/datum/reagent/medicine/meralyne
+	name = "Meralyne"
+	id = "meralyne"
+	description = "Meralyne is a concentrated form of bicardine and can be used to treat extensive blunt trauma."
+	color = "#E6666C"
+	overdose_threshold = REAGENTS_OVERDOSE*0.5
+	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL*0.5
+	scannable = TRUE
+
+/datum/reagent/medicine/meralyne/on_mob_life(mob/living/M, alien)
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]:
```suggestion
/datum/reagent/medicine/meralyne/on_mob_life(mob/living/L, alien)
```

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 3]
```diff
@@ -627,6 +627,27 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
...
+	M.heal_limb_damage(4 * REM, 0)
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]:
```suggestion
	L.heal_limb_damage(4 * REM, 0)
```

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 4]
```diff
@@ -627,6 +627,27 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
...
+	M.reagent_move_delay_modifier += 1
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]:
```suggestion
	L.reagent_move_delay_modifier += 1
```

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 5]
```diff
@@ -627,6 +627,27 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
...
+	
+	return ..()
+
+/datum/reagent/medicine/meralyne/overdose_process(mob/living/M, alien)
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]:
```suggestion
/datum/reagent/medicine/meralyne/overdose_process(mob/living/L, alien)
```

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 6]
```diff
@@ -627,6 +627,27 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
...
+	M.apply_damage(2, BURN)
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]:
```suggestion
	L.apply_damage(2, BURN)
```

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 7]
```diff
@@ -627,6 +627,27 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
...
+
+/datum/reagent/medicine/meralyne/overdose_crit_process(mob/living/M, alien)
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]:
```suggestion
/datum/reagent/medicine/meralyne/overdose_crit_process(mob/living/L, alien)
```

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 8]
```diff
@@ -627,6 +627,27 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
...
+	M.apply_damages(2, 6, 4)
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reagents/medical.dm]:
```suggestion
	L.apply_damages(2, 6, 4)
```

**Comment by AdamElTablawy** [comment]
i find stacking combat healing less of an issue than it is that meralyne is just bicardine but better with no differences in how it functions and trivially easy to make

instead of, say, on /tg/, where there are multiple burn and brute healing chemicals but some of them are better than others and each of them only functions in a particular way (on touch, on ingest, on inject, side effect like hiding health hud or only working at certain health points) this is just better healing and that's where it leaves off, and it's easier to mass produce

ideally a doctor should be able to give a marine stacking healing, but it shouldn't just be 'better version of the other thing no reason not to take it.' if meralyne only worked below a certain health percentage, that might make it better for backline use-it would revive seriously injured patients but be wasted just slamming it in a bicardine/meralyne pill normally since otherwise you could just use a regular bicardine pill. could make a downside like making health hud appear 100% at all times while it is in you so you can't evaluate your own health aside from movement speed and pain messages, et cetera.

anything other than a slowdown that relegates it to 'never used medicine' because someone at the back of the line will be healed regardless of if you give them ambrosia of the gods or goat piss

**Comment by Rohesie** [comment]
Something seems to have gone wrong with the rebase, and it's now touching on ` tmp/iconCache.sav` without making changes, and adding an already existing chemical instead of modifying it.

**Comment by Lainwright-Lazuli** [comment]
Sorry, I was out for awhile. Something came up: I'll look into the code ASAP.

**Comment by Lainwright-Lazuli** [comment]
Hmm, thanks for the ideas Adam! I'll certainly think about these kinds of changes.

File: code/modules/reagents/chemistry_reactions/medical.dm [hunk 9]
```diff
@@ -37,6 +37,12 @@
 	results = list("bicaridine" = 2)
 	required_reagents = list("inaprovaline" = 1, "carbon" = 1)
 
+/datum/chemical_reaction/meralyne
```
Comment (DominikPanic) [inline code/modules/reagents/chemistry_reactions/medical.dm]: Meralyne already exists, likely a bad branch.

**Comment by Rohesie** [comment]
Closing this PR. The feature is good, maybe when the author returns we can reopen it.

**Pull request closed without merging**